pennylane/measurements.py

```diff
@@ -288,6 +291,98 @@ def shape(self, device=None):
             + f"{self.return_type}."
         )
 
+    @functools.lru_cache()
+    def shape_new(self, device=None):
```

Comment:
Let's make this private.
```suggestion
    def _shape_new(self, device=None):
```